addons/nightvision/functions/fnc_blending.sqf

```diff
@@ -18,44 +18,33 @@
 //IGNORE_PRIVATE_WARNING ["_unit", "_weapon", "_muzzle", "_mode", "_ammo", "_magazine", "_projectile", "_vehicle", "_gunner", "_turret"];
 TRACE_10("firedEH:",_unit, _weapon, _muzzle, _mode, _ammo, _magazine, _projectile, _vehicle, _gunner, _turret);
 
-private "_player";
-_player = ACE_player;
+if (_gunner != ACE_player) exitWith {};
 
 //If we're not in NVG mode, or it's a grenade, exit
-if (((currentVisionMode _player) != 1) || {_weapon == "Throw"}) exitWith {};
+if (currentVisionMode _gunner != 1 || {toLower _weapon in ["throw", "put"]}) exitWith {};
 
-private["_darkness", "_nvgBrightnessCoef", "_silencer", "_visibleFire", "_visibleFireCoef", "_visibleFireTime", "_visibleFireTimeCoef"];
+private _silencer = _gunner weaponAccessories _weapon select 0;
 
-_silencer = switch (_weapon) do {
-    case (primaryWeapon _player): {(primaryWeaponItems _player) select 0};
-    case (secondaryWeapon _player): {(secondaryWeaponItems _player) select 0};
-    case (handgunWeapon _player): {(handgunItems _player) select 0};
-    default {""};
-};
+private _visibleFireCoef = 1;
+private _visibleFireTimeCoef = 1;
 
-_visibleFireCoef = 1;
-_visibleFireTimeCoef = 1;
 if (_silencer != "") then {
     _visibleFireCoef = getNumber (configFile >> "CfgWeapons" >> _silencer >> "ItemInfo" >> "AmmoCoef" >> "visibleFire");
     _visibleFireTimeCoef = getNumber (configFile >> "CfgWeapons" >> _silencer >> "ItemInfo" >> "AmmoCoef" >> "visibleFireTime");
 };
 
-_visibleFire = getNumber (configFile >> "CfgAmmo" >> _ammo >> "visibleFire");
-_visibleFireTime = getNumber (configFile >> "CfgAmmo" >> _ammo >> "visibleFireTime");
-
-_nvgBrightnessCoef = 1 + (_player getVariable [QGVAR(NVGBrightness), 0]) / 4;
-
-_fnc_isTracer = {
-    private ["_indexShot", "_lastRoundsTracer", "_tracersEvery"];
+private _visibleFire = getNumber (configFile >> "CfgAmmo" >> _ammo >> "visibleFire");
+private _visibleFireTime = getNumber (configFile >> "CfgAmmo" >> _ammo >> "visibleFireTime");
 
+private _fnc_isTracer = {
     if (getNumber (configFile >> "CfgAmmo" >> _ammo >> "nvgOnly") > 0) exitWith {false};
 
-    _indexShot = (_player ammo _weapon) + 1;
+    private _indexShot = (_gunner ammo _weapon) + 1;
```

Comment:
tested and `_unit ammo _weapon` as well as `_gunner ammo _weapon` report the same correct value of the vehicle